packages/client-core/src/common/services/FileThumbnailJobState.tsx

```diff
@@ -170,89 +212,103 @@ const uploadThumbnail = async (src: string, projectName: string, blob: Blob | nu
     ;(e) => console.error(e)
   }
 }
+const useGenerateHelper = (
+  files: readonly FileBrowserContentType[],
+  filterKey: (file: FileBrowserContentType) => string | undefined,
+  queryConditions: Record<string, any>,
+  jobType: 'thumbnail' | 'dimension' = 'thumbnail'
+) => {
+  const jobState = useMutableState(FileThumbnailJobState)
+  const seenResources = jobState.seenResources[jobType]
+  const fileList = files
+    .map(filterKey)
+    .filter((key): key is string => key !== undefined)
+    .filter((key) => !seenResources.value.includes(key))
```

Comment:
both thumbnail and dimension jobs are sharing `seenResources`. They need to have different seenResources objects for each job type so they don't get confused